source/main.cpp

```diff
@@ -13,44 +16,53 @@
 #include "parser/parser.h"
 
 int main(int argc, char ** argv) {
+  // Install the segfault handler with backtrace
+  signal(SIGSEGV, handler);
 
-// The valid command line usage is "./mc-mini <parameter file>". Otherwise, throw an exception.
-  if (argc == 1) {
-    throw std::invalid_argument("usage: " + std::string{argv[0]} + " <parameter file>.");
-  }
+  try {
+    // The valid command line usage is "./mc-mini <parameter file>". Otherwise, throw an exception.
+    if (argc == 1) {
+      THROW_WITH_TRACE(InvalidArgument() <<
+              errmsg_info("usage: " + static_cast<std::string>(argv[0]) + " <parameter file>."));
```

Comment:
What is the difference between `std::string{argv[0]}` and `static_cast<std::string>(argv[0])`? The first way seems more natural to me: perhaps there is some advantage to the second way?